packages/yourloops/lib/data/api.ts

```diff
@@ -124,6 +124,10 @@ export async function getPatientDataRange(session: Session, patient: IUser): Pro
     }
     return dataRange;
   } else if (response.status === HttpStatus.StatusNotFound) {
+    if (appConfig.CBG_BUCKETS_ENABLED) {
+      // The v2 route return nothing
+      return null;
+    }
     try {
       const text = await response.text();
       if (text.length > 0) {
```

Comment:
Do you know if it can happen for the v1 route, not to go in this if condition?
Because if yes, the best fix might be to always return null when `response.status === HttpStatus.StatusNotFound`.
Otherwise, I don't know much about this code. I have tested this fix and it works. 
If you know someone more competent to review it, I think that it would be a good idea

Reply:
I understand all of this. :wink: 
I checked the code of the v1 route. It cannot return a StatusNotFound without the body having a content. 
I suggest to delete everything between line 127-142 and simply return null instead.

Reply:
Yes, but this code is for if the route is removed, disabled, or a miss-configuration of the nginx router in k8s
But sure I can remove it